jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -1,68 +1,146 @@
+    public void showHide(View view) {
 
-            // Edit the text view (show written message).
-            TextView textView = (TextView)
-                    display_message_fragment.getView().findViewById(R.id.display_message);
-            textView.setText(message);
+        // If the displaying fragment has not been initialized return.
+        if (display_message_fragment == null) {
+            return;
+        }
 
-        // If orientation is in portrait start new activity.
-        } else {
-            // Create intent.
-            Intent intent = new Intent(this, DisplayMessageActivity.class);
+        // Setup FragmentManager and FragmentTransaction.
+        View fragment_container = findViewById(R.id.fragment_container);
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction ft = fragmentManager.beginTransaction();
+        ft.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out);
+
+
+        // Hide Fragment if it is visible.
+        if (display_message_fragment.isVisible()) {
+            ft.hide(display_message_fragment).commitNow();
+
+            isDisplayMessageFragmentVisible = false;
+
+            // Inform the main fragment of the change
+            main_fragment.setButtonText();
+
+            // Set width of fragment layout container to 0.
+            changeFragmentLayout(fragment_container, 0);
+            return;
+        }
+
+        // Show Fragment if it is hidden.
+        if (display_message_fragment.isHidden()) {
+            ft.show(display_message_fragment).commitNow();
+
+            isDisplayMessageFragmentVisible = true;
 
-            // Add (user) entered message to the created intent.
-            intent.putExtra(EXTRA_MESSAGE, message);
+            // Inform the main fragment of the change
+            main_fragment.setButtonText();
 
-            // Start an instance of the DisplayMessageActivity specified by the intent.
-            startActivity(intent);
+            // Set width of fragment layout container to fill half the screen.
+            changeFragmentLayout(fragment_container, ViewGroup.LayoutParams.MATCH_PARENT);
+            return;
         }
 
+    }
+
+    /** Helper function to change the layout of the fragment container */
+    private void changeFragmentLayout(View fragment_layout, int newLayoutParam) {
+
+        // Get the current Layout parameters.
+        ViewGroup.LayoutParams layoutParams = fragment_layout.getLayoutParams();
+
+        // Set the width or height of the layout that contains the fragment.
+        if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
+            layoutParams.height = newLayoutParam;
+        } else {
+            layoutParams.width = newLayoutParam;
+        }
+
+        // Set new Layout parameters.
+        fragment_layout.setLayoutParams(layoutParams);
+    }
 
+    /** Retrieves the isDisplayMessageFragmentVisible boolean. */
```

Comment:
Consider whether this is good documentation. I see you add several comments that function as self-notes for you to learn and remember - which is great for now. Moving ahead think of comments as as way to explain code that is not self-explanatory and to document. For example, for this method you would want some documentation, so that when using the method it is described in plain English what `isDisplayMessageFragmentVisible` is. Currently your comments says nothing more than what the method signature already says.